sklearn/cross_decomposition/tests/test_pls.py

```diff
@@ -148,7 +148,11 @@ def test_sanity_check_pls_regression_constant_column_Y():
 
     x_loadings_sign_flip = np.sign(expected_x_loadings / pls.x_loadings_)
     x_weights_sign_flip = np.sign(expected_x_weights / pls.x_weights_)
-    y_loadings_sign_flip = np.sign(expected_y_loadings / pls.y_loadings_)
+
+    y_loadings_sign_flip = np.sign(expected_y_loadings[1:] /
+                                   pls.y_loadings_[1:])
+    y_loadings_sign_flip = np.insert(y_loadings_sign_flip, 0, 0, axis=0)
```

Comment:
I'd remove that and below use

```
     assert_array_equal(x_loadings_sign_flip[1:], y_loadings_sign_flip) # <- removed the [:1] here
```

Also let's add a small comment right above like "we ignore the first full-zeros row for y"